pstoreds/cache.go

```diff
@@ -31,3 +34,7 @@ func (*noopCache) Contains(key interface{}) bool {
 func (*noopCache) Peek(key interface{}) (value interface{}, ok bool) {
```

Comment:
Are you sure that everything will be working with `Peek` returning false. It seemed to me that `flusher` will not flush entries that are out of the cache.

Reply:
Removed the _async_ flush altogether. The complexity of keeping it was becoming high and it was really premature optimisation.

Reason: with a noop cache, we cannot know if the record queued for async flush is the latest state, so we can easily incur in a dirty write if we (1) loaded the record and it contained expired entries, therefore we queued a flush; (2) we modified the record and persisted it synchronously; (3) we process the async flush.